presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/HiveMetastoreClientFactory.java

```diff
@@ -49,12 +72,138 @@ public HiveMetastoreClientFactory(
     @Inject
     public HiveMetastoreClientFactory(MetastoreClientConfig metastoreClientConfig, HiveMetastoreAuthentication metastoreAuthentication)
     {
-        this(Optional.empty(), Optional.ofNullable(metastoreClientConfig.getMetastoreSocksProxy()), metastoreClientConfig.getMetastoreTimeout(), metastoreAuthentication);
+        this(metastoreSslContext(metastoreClientConfig.getMetastoreTlsEnabled(), Optional.ofNullable(metastoreClientConfig.getMetastoreTlsKeystorePath()), Optional.ofNullable(metastoreClientConfig.getMetastoreTlsKeystorePassword()), Optional.ofNullable(metastoreClientConfig.getMetastoreTlsTruststorePath()), Optional.ofNullable(metastoreClientConfig.getMetastoreTlsTruststorePassword())), Optional.ofNullable(metastoreClientConfig.getMetastoreSocksProxy()), metastoreClientConfig.getMetastoreTimeout(), metastoreAuthentication);
     }
 
     public HiveMetastoreClient create(HostAndPort address, Optional<String> token)
             throws TTransportException
     {
         return new ThriftHiveMetastoreClient(Transport.create(address, sslContext, socksProxy, timeoutMillis, metastoreAuthentication, token));
     }
+
+    /**
+     * Reads the truststore and keystore and returns the SSLContext
+     * @param metastoreTlsEnabled
+     * @param metastoreKeyStorePath
+     * @param metastoreKeyStorePassword
+     * @param metastoreTrustStorePath
+     * @param metastoreTrustStorePassword
+     * @return SSLContext
+     */
+    private static Optional<SSLContext> metastoreSslContext(boolean metastoreTlsEnabled, Optional<File> metastoreKeyStorePath, Optional<String> metastoreKeyStorePassword, Optional<File> metastoreTrustStorePath, Optional<String> metastoreTrustStorePassword)
+    {
+        if (!metastoreTlsEnabled || (!metastoreKeyStorePath.isPresent() && !metastoreTrustStorePath.isPresent())) {
```

Comment:
I think we can simply this here -
```suggestion
        if (!metastoreTlsEnabled || (metastoreKeyStorePath.isEmpty() && metastoreTrustStorePath.isEmpty()) {

```

Reply:
@agrawalreetika [Optional#isEmpty()](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html#isEmpty()) is a Java 11 method. Not available in java 8